internal/lsp/server.go

```diff
@@ -2029,45 +2011,84 @@ func (l *LanguageServer) handleInitialize(ctx context.Context, params types.Init
 		l.log.Message("failed to cache enabled rules: %s", err)
 	}
 
-	if l.workspaceRootURI != "" {
-		workspaceRootPath := l.workspacePath()
+	if params.RootURI != "" {
```

Comment:
Should we also look for this in the workspaceFolders param?
The spec suggests that's where it should be, and we've mostly been lucky that clients don't seem to honor that https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#initializeParams

(and it's easy to see why they don't given how poorly that is documented)

Not suggesting we handle multiple workspaces as part of this, but we could perhaps just check if there's _something_ there, and if there is we go with the first one (if nothing else was given).

Reply:
yeah that's fair, I have something in there now